aiohttp/streams.py

```diff
@@ -321,9 +327,6 @@ async def readuntil(self, separator: bytes = b"\n") -> bytes:
         if seplen == 0:
             raise ValueError("Separator should be at least one-byte string")
 
-        if self._exception is not None:
-            raise self._exception
```

Comment:
I think you cannot just drop the check here and below but should lookup for `self._waiter` has an exception and re-raise it if present.

Reply:
Yep, but if we leave `self._exception` circular reference will be still exist

Reply:
`weakref` can help, but as i understood it doesn't work well with `Exception`, because we can't raise `weakref`

Reply:
ok, will fix it

Reply:
Done